cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/TransformExecutorFactory.java

```diff
@@ -131,7 +138,14 @@ public TransformExecutorFactory(JobContext hadoopContext, PipelinePluginInstanti
       return;
     }
 
-    addTransformation(pipeline, stageName, transformations, transformErrorSinkMap);
+    try {
+      addTransformation(pipeline, stageName, transformations, transformErrorSinkMap);
+    } catch (Exception e) {
+      // Catch the Exception to generate a User Error Log for the Pipeline
+      PIPELINELOG.error("Transform Stage '{}' failed. Please review your Pipeline configuration" +
```

Comment:
@anew This happens [e.g.] when initializing Mapper for a certain transformation. Not sure if we can call it "Failed to initialize" for the plugin. I guess we could, please let me know what you think. 

	at co.cask.cdap.etl.batch.TransformExecutorFactory.addTransformation(TransformExecutorFactory.java:179) [cdap-etl-batch-4.2.0-SNAPSHOT.jar:na]
	at co.cask.cdap.etl.batch.TransformExecutorFactory.setPipeTransformDetail(TransformExecutorFactory.java:142) [cdap-etl-batch-4.2.0-SNAPSHOT.jar:na]
	at co.cask.cdap.etl.batch.TransformExecutorFactory.setPipeTransformDetail(TransformExecutorFactory.java:151) [cdap-etl-batch-4.2.0-SNAPSHOT.jar:na]
	at co.cask.cdap.etl.batch.TransformExecutorFactory.setPipeTransformDetail(TransformExecutorFactory.java:151) [cdap-etl-batch-4.2.0-SNAPSHOT.jar:na]
	at co.cask.cdap.etl.batch.TransformExecutorFactory.create(TransformExecutorFactory.java:113) [cdap-etl-batch-4.2.0-SNAPSHOT.jar:na]
	at co.cask.cdap.etl.batch.mapreduce.TransformRunner.<init>(TransformRunner.java:115) [cdap-etl-batch-4.2.0-SNAPSHOT.jar:na]
	at co.cask.cdap.etl.batch.mapreduce.ETLMapReduce$ETLMapper.initialize(ETLMapReduce.java:362) [cdap-etl-batch-4.2.0-SNAPSHOT.jar:na]
	at co.cask.cdap.etl.batch.mapreduce.ETLMapReduce$ETLMapper.initialize(ETLMapReduce.java:348) [cdap-etl-batch-4.2.0-SNAPSHOT.jar:na]
	at co.cask.cdap.internal.app.runtime.batch.MapperWrapper.run(MapperWrapper.java:108) [na:na]